File: src/test/java/at/deckweiss/logging/LogParamsTest.java
package at.deckweiss.logging;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.tuple;

class LogParamsTest {

    @Test
    void givenNoLogParams_whenAddingNewParam_thenParamIsAdded() {
        LogParams logParams = new LogParams();
        logParams.add("my new log object", "asdf");

        assertThat(logParams.toArray()).extracting("paramName", "paramValue")
                                       .containsExactly(tuple("my new log object", "asdf"));
    }

    @Test
    void givenNoLogParams_whenAddingNewNullParam_thenParamIsAdded() {
        LogParams logParams = new LogParams();
        logParams.add("my new log object", null);

        assertThat(logParams.toArray()).extracting("paramName", "paramValue")
                                       .containsExactly(tuple("my new log object", null));
    }

    @Test
    void givenNoLogParams_whenAddingIfNotNullNewParam_thenParamIsAdded() {
        LogParams logParams = new LogParams();
        logParams.addIfNotNull("my new log object", "asdf");

        assertThat(logParams.toArray()).extracting("paramName", "paramValue")
                                       .containsExactly(tuple("my new log object", "asdf"));
    }

    @Test
    void givenNoLogParams_whenAddingIfNotNullNewNullParam_thenParamIsNotAdded() {
        LogParams logParams = new LogParams();
        logParams.addIfNotNull("my new log object", null);

        assertThat(logParams.toArray()).isEmpty();
    }

    @Test
    void givenMultipleLogParam_whenAddingMultipleParams_themParamsAreAdded() {
        LogParam param1 = new LogParam("param1", "test1");
        LogParam param2 = new LogParam("param2", "test2");

        LogParams params = new LogParams().add(param1, param2);

        assertThat(params.toArray()).extracting("paramName", "paramValue")
                                    .containsExactly(
                                            tuple("param1", "test1"),
                                            tuple("param2", "test2")
                                    );
    }
}

File: src/test/kotlin/at/deckweiss/logging/LogParamsTest.kt
package at.deckweiss.logging

import org.junit.jupiter.api.Test

class LogParamsTest {

    @Test
    fun `Given no LogParams, when adding new param, then param is added`() {
        val logParams: LogParams = LogParams()
        logParams.add("my new log object", "asdf")

        assert(logParams.toParamString() == "[my new log object=\"asdf\"]")
    }

    @Test
    fun `Given no LogParams, when adding new null param, then param is added`() {
        val logParams: LogParams = LogParams()
        logParams.add("my new log object", null)

        assert(logParams.toParamString() == "[my new log object=\"null\"]")
    }

    @Test
    fun `Given no LogParams, when adding new param with AddIfNotNull, then param is added`() {
        val logParams: LogParams = LogParams()
        logParams.addIfNotNull("my new log object", "asdf")

        assert(logParams.toParamString() == "[my new log object=\"asdf\"]")
    }

    @Test
    fun `Given no LogParams, when adding new null param with AddIfNotNull, then param is not added`() {
        val logParams: LogParams = LogParams()
        logParams.addIfNotNull("my new log object", null)

        assert(logParams.toParamString() == "")
    }

    @Test
    fun `Given multiple LogParams, when adding multiple params, then params are added`() {
        val logParams: LogParams = LogParams()

        logParams.add("param1", "test1").add("param2", "test2")

        assert(logParams.toParamString() == "[param1=\"test1\", param2=\"test2\"]")
    }
}